### ISSUE TITLE
Lectern with atlas - null pointer

### ISSUE THREAD

**Issue by dsnarski33** [Pepperoni-Jabroni/MapAtlases#105]
Hi. Maybe this can be of use for you. Thanks for the mod and all the work that goes into it. I really like the concept you have in place. I actually wanted a Forge port of this, but was having issues finding a reliable one for what I wanted. I started tinkering with putting it together myself, and came across this issue that does seem to still be in your latest version. There looks to be a comment also to this end, "TODO: Sometimes throws bc of null when opening Lectern". In `MapAtlasItem.createMenu()` the top line referencing `getAtlasFromLookingLectern()` can fail to find the atlas if you hit the very top of the lectern. The lecterns size extends past the boundary of one block, and the raycast could then be checking the wrong block position via `HitResult.getPos()`. I think you could instead confirm `HitResult.Type.BLOCK` and cast to `BlockHitResult` and use `blockPos`. Seems right to me, but maybe theres a better solution. I am new-ish to learning the Minecraft modding processes. Hope this helps. Thanks.

**Comment by Pepperoni-Jabroni** [comment]
Yeah I meant to address this earlier, but the current Lectern implementation is the rest of a pretty egregious weekend  of hacking something together I'm afraid. Thanks for taking a look at the root cause! I'll take a look and see if this is the case. Appreciate you reporting this!
